src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,305 @@
+     * Note: registration based on other user attributes (e.g. email, name, login) is currently NOT supported
+     * <p>
+     * This method first tries to find the student in the internal Artemis user database (because the user is most probably already using Artemis).
+     * In case the user cannot be found, we additionally search the (TUM) LDAP in case it is configured properly.
+     *
+     * @param courseId        the id of the course
+     * @param tutorialGroupId the id of the tutorial group to which the users should be registered to
+     * @param studentDtos     the list of students (with at least registration number) who should get access to the course
+     * @return the list of students who could not be registered for the tutorial group, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/register-multiple")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<List<StudentDTO>> registerMultipleStudentsToTutorialGroup(@PathVariable Long courseId, @PathVariable Long tutorialGroupId,
+            @RequestBody List<StudentDTO> studentDtos) {
+        log.debug("REST request to register {} to tutorial group {} of course {}", studentDtos, tutorialGroupId, courseId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        List<StudentDTO> notFoundStudentsDtos = tutorialGroupService.registerMultipleStudents(courseId, tutorialGroupId, studentDtos);
+        return ResponseEntity.ok().body(notFoundStudentsDtos);
+    }
+
+    private void checkTutorialCourseIdMatchesPathId(Long courseIdFromPath, TutorialGroup tutorialGroupFromDatabase) {
+        if (!tutorialGroupFromDatabase.getCourse().getId().equals(courseIdFromPath)) {
+            throw new ConflictException("The tutorial group does not belong to the correct course", "TutorialGroup", "tutorialGroupWrongCourse");
+        }
+    }
+
+    private void trimStringFields(TutorialGroup tutorialGroup) {
+        if (tutorialGroup.getTitle() != null) {
+            tutorialGroup.setTitle(tutorialGroup.getTitle().trim());
+        }
+        if (tutorialGroup.getAdditionalInformation() != null) {
+            tutorialGroup.setAdditionalInformation(tutorialGroup.getAdditionalInformation().trim());
+        }
+    }
+
+    private void checkTitleIsUnique(TutorialGroup tutorialGroup) {
+        if (tutorialGroupRepository.findAllByCourseId(tutorialGroup.getCourse().getId()).stream().map(TutorialGroup::getTitle)
```

Comment:
This could be moved into a special SQL query, which would greatly improve the performance, as you would not have to fetch all existing tutor groups of the course.

Reply:
Very good idea! Done in  f2af95843922f41c3d4ef62efd17aa3f02016222 :)